### PR TITLE
[READY] - added vim config to hide mouse and locale for military time

### PR DESCRIPTION
**Pull request by fossadmin**
## Description of PR

Fixes: #694 

Added vim config to disable mouse plus plugin for nix syntax highlighting.

Also, added locale settings for 24h military time.

### CODE REVIEW

File: nix/machines/_common/base.nix [hunk 1]
```diff
@@ -36,15 +36,41 @@
...
+  i18n.extraLocaleSettings = {
+    LC_ADDRESS = "en_US.UTF-8";
+    LC_IDENTIFICATION = "en_US.UTF-8";
+    LC_MEASUREMENT = "en_US.UTF-8";
+    LC_MONETARY = "en_US.UTF-8";
+    LC_NAME = "en_US.UTF-8";
+    LC_NUMERIC = "en_US.UTF-8";
+    LC_PAPER = "en_US.UTF-8";
+    LC_TELEPHONE = "en_US.UTF-8";
+    LC_TIME = "C.UTF-8";
```
Comment (sarcasticadmin) [inline nix/machines/_common/base.nix:71]: @owendelong can you confirm how to you expected to set military time?

The `LC_TIME` locale setting seems to correctly set the `date` output, but wanted confirmation

**Review by kylerisse — COMMENTED** [review]
Hi @fossadmin . Looks like CI is failing from the new drive by formatter implementation...

https://github.com/socallinuxexpo/scale-network/actions/runs/10984414775/job/30494982929?pr=774

please run `nixfmt` against `/nix/machines/_common/base.nix` and rebase

**Comment by fossadmin** [comment]
Hey @kylerisse , just pushed reformatted update through.

**Comment by fossadmin** [comment]
Looks like reformatting with nixfmt is what caused "nix checks" to fail. Manually reformatted based on the errors and "nix checks" is now successful.

**Comment by fossadmin** [comment]
FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.

**Comment by kylerisse** [comment]
> FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.

Not sure. This was just merged Saturday night with no documented guidelines, so you’re the first victim. Best to get comment from for @djacu and @sarcasticadmin .

**Comment by sarcasticadmin** [comment]
> FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.

Since its part of the flake.nix `formatter` you should be able to just run `nix fmt` from inside the scale-network repo and itll call the same formatters as CI and format to appropriate files in the repo. We definitely should document this.

**Comment by fossadmin** [comment]
> > FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.
> 
> Since its part of the flake.nix `formatter` you should be able to just run `nix fmt` from inside the scale-network repo and itll call the same formatters as CI and format to appropriate files in the repo. We definitely should document this.

@sarcasticadmin maybe a shell.nix file with all the tools needed so we can just run nix-shell in the repo and have a similar dev environment with everything we need?

I just ran `nix fmt` to test it out and got the following error:
```
error:
       … in the left operand of the update (//) operator

         at «string»:56:13:

           55|             # This is shadowed in the next //
           56|             // sourceInfo
             |             ^
           57|             // {

       … in the left operand of the update (//) operator

         at /nix/store/zazfm8nliz6ar3310x1cf9ynlf3ysl5f-source/flake.nix:31:5:

           30|     })
           31|     // {
             |     ^
           32|       formatter = import ./nix/formatter inputs;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: unable to download 'https://api.github.com/repos/hercules-ci/flake-parts/tarball/7f53fdb7bdc5bb237da7fefef12d099e4fd611ca': HTTP error 401

       response body:

       {
         "message": "Bad credentials",
         "documentation_url": "https://docs.github.com/rest",
         "status": "401"
       }
```

**Comment by kylerisse** [comment]
> > > FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.

Thanks @sarcasticadmin for clearing up the confusion. 

Hi @fossadmin you can get the equivalent of a `nix-shell` with `nix develop`

```
nix develop
nix fmt
```

**Comment by kylerisse** [comment]
@fossadmin next request is to please squash your commits. Since you have 4 the process should be:

```
git rebase -i HEAD~4
```

In the editor, replace the word `pick` on 2nd through 4th commit lines with `s` for squash. Save the file buffer. You will then be in your editor again with the commit messages. I suggest you just remove all lines below the first so the message is the same as the 1st commit. You should then see just 1 commit in your git log with all changes included. You will have to force push the branch (git push -f) to update this PR but should now see only 1 clean commit. I'll ping you on `that chat thing we use for that group that we're in` also so we can discuss there if you have any questions.

**Comment by fossadmin** [comment]
> > > > FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.
> 
> Thanks @sarcasticadmin for clearing up the confusion. 
> 
> Hi @fossadmin you can get the equivalent of a `nix-shell` with `nix develop`
> 
> ```
> nix develop
> nix fmt
> ``` 
> 

@kylerisse I tried `nix develop` and got the same error as the one in my previous comment. I think there's something wrong with my laptop. Will reinstall NixOS in the morning, in the process of backing up my data now.

**Comment by fossadmin** [comment]
> @fossadmin next request is to please squash your commits. Since you have 4 the process should be:
> 
> ```
> git rebase -i HEAD~4
> ```
> 
> In the editor, replace the word `pick` on 2nd through 4th commit lines with `s` for squash. Save the file buffer. You will then be in your editor again with the commit messages. I suggest you just remove all lines below the first so the message is the same as the 1st commit. You should then see just 1 commit in your git log with all changes included. You will have to force push the branch (git push -f) to update this PR but should now see only 1 clean commit. I'll ping you on `that chat thing we use for that group that we're in` also so we can discuss there if you have any questions.

@kylerisse your instructions were perfect, thanks! Looks like it's down to 1 commit now.

**Comment by djacu** [comment]
> > > FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.
> > 
> > 
> > Since its part of the flake.nix `formatter` you should be able to just run `nix fmt` from inside the scale-network repo and itll call the same formatters as CI and format to appropriate files in the repo. We definitely should document this.
> 
> @sarcasticadmin maybe a shell.nix file with all the tools needed so we can just run nix-shell in the repo and have a similar dev environment with everything we need?
> 
> I just ran `nix fmt` to test the formatter you mentioned and got the following error:
> 
> ```
> error:
>        … in the left operand of the update (//) operator
> 
>          at «string»:56:13:
> 
>            55|             # This is shadowed in the next //
>            56|             // sourceInfo
>              |             ^
>            57|             // {
> 
>        … in the left operand of the update (//) operator
> 
>          at /nix/store/zazfm8nliz6ar3310x1cf9ynlf3ysl5f-source/flake.nix:31:5:
> 
>            30|     })
>            31|     // {
>              |     ^
>            32|       formatter = import ./nix/formatter inputs;
> 
>        (stack trace truncated; use '--show-trace' to show the full trace)
> 
>        error: unable to download 'https://api.github.com/repos/hercules-ci/flake-parts/tarball/7f53fdb7bdc5bb237da7fefef12d099e4fd611ca': HTTP error 401
> 
>        response body:
> 
>        {
>          "message": "Bad credentials",
>          "documentation_url": "https://docs.github.com/rest",
>          "status": "401"
>        }
> ```

This doesn't look like a nix or dev environment problem but possibly that you are getting rate limited by GH. Do you have a PAT setup?

**Comment by fossadmin** [comment]
Hey @djacu , I'm using SSH keys but just checked my PAT setup and it says it expired on Sept. 6th. I'll update it in the morning and see if that fixes my problem. Still probably need to reinstall NixOS due to some other issues with the wofi app launcher for wayland throwing this error:
```
segmentation fault (core dumped)
```

**Comment by sarcasticadmin** [comment]
> > > FYI, I do not have the nixfmt command on my system but I'm running NixOS unstable and ran `nix-shell -p nixfmt` to run it on the base.nix file. Not sure if that is what caused the issues.
> > 
> > 
> > Since its part of the flake.nix `formatter` you should be able to just run `nix fmt` from inside the scale-network repo and itll call the same formatters as CI and format to appropriate files in the repo. We definitely should document this.
> 
> @sarcasticadmin maybe a shell.nix file with all the tools needed so we can just run nix-shell in the repo and have a similar dev environment with everything we need?
> 
> I just ran `nix fmt` to test the formatter you mentioned and got the following error:
> 
> ```
> error:
>        … in the left operand of the update (//) operator
> 
>          at «string»:56:13:
> 
>            55|             # This is shadowed in the next //
>            56|             // sourceInfo
>              |             ^
>            57|             // {
> 
>        … in the left operand of the update (//) operator
> 
>          at /nix/store/zazfm8nliz6ar3310x1cf9ynlf3ysl5f-source/flake.nix:31:5:
> 
>            30|     })
>            31|     // {
>              |     ^
>            32|       formatter = import ./nix/formatter inputs;
> 
>        (stack trace truncated; use '--show-trace' to show the full trace)
> 
>        error: unable to download 'https://api.github.com/repos/hercules-ci/flake-parts/tarball/7f53fdb7bdc5bb237da7fefef12d099e4fd611ca': HTTP error 401
> 
>        response body:
> 
>        {
>          "message": "Bad credentials",
>          "documentation_url": "https://docs.github.com/rest",
>          "status": "401"
>        }
> ```

That was the same error we say while were at nullspace. Its unclear why you'd be setting a bad credentials error since you dont need credentials to download the source: https://api.github.com/repos/hercules-ci/flake-parts/tarball/7f53fdb7bdc5bb237da7fefef12d099e4fd611ca (Im able to download it successfully). Are you using some kind of shared VPN or proxy? I like djacu's recommendation of trying to install a GH PAT to see if that helps, but its still odd that you get a 401 from the gitub api. 

Another thing to check is that you dont have a `~/.netrc` or similar file with credentials to github since that could be causing the 401. https://nixos.wiki/wiki/Enterprise mentions setting where else this could be configured for nixos.

**Comment by fossadmin** [comment]
@sarcasticadmin yeah, it's really weird. I don't see any .netrc files and not using a VPN or proxy. I'm hoping a fresh install does the trick but I'll try the PAT as well.

**Review by kylerisse — COMMENTED** [review]
@fossadmin thanks for rebasing and working through the formatting. PR LGTM! We should have @owendelong review before merging as originally noted.

**Comment by djacu** [comment]
I don't completely understand it either but searching for some keywords returned a bunch of gh rate limiting discussions. If you can download the file in your browser by following the link then it's possibly something else

**Comment by fossadmin** [comment]
> I don't completely understand it either but searching for some keywords returned a bunch of gh rate limiting discussions. If you can download the file in your browser by following the link then it's possibly something else

It was really weird because I was able to download the file via that link in my browser. I reinstalled NixOS this morning and seems to be working fine now.

**Comment by sarcasticadmin** [comment]
Tested this by modifying the existing `loghost` vmTest and importing `machines/_common/base.nix`:

ss-202409271727422236
ss-202409271727422225
ss-202409271727422212

The configuration looks like its loading as expected for `vim` and `date`

File: nix/machines/_common/base.nix [hunk 2]
```diff
@@ -36,15 +36,44 @@
...
     unixtools.nettools
-    vim
     wget
+    ((vim_configurable.override { }).customize {
+      name = "vim";
+      # Install plugins for example for syntax highlighting of nix files
+      vimrcConfig.packages.myplugins = with pkgs.vimPlugins; {
+        start = [
+          vim-nix
+          vim-lastplace
```
Comment (sarcasticadmin) [inline nix/machines/_common/base.nix:46]: lets get rid of `vim-lastplace` I dont think we'd have much use for it on the server

File: nix/machines/_common/base.nix [hunk 3]
```diff
@@ -36,15 +36,44 @@
...
+    })
   ];
 
   # Purge nano from being the default
   environment.variables = {
     EDITOR = "vim";
   };
 
+  # set 24h military time
+  i18n.extraLocaleSettings = {
```
Comment (sarcasticadmin) [inline nix/machines/_common/base.nix:65]: Can we remove all `LC_*` config and just leave the `LC_TIME`? `LC_TIME` is the only one that were concerned with at the moment

**Review by sarcasticadmin — COMMENTED** [review]
If you can fixup the changes once your done, then Ill merge :shipit:

**Comment by fossadmin** [comment]
> If you can fixup the changes once your done, then Ill merge :shipit:

@sarcasticadmin done!

**Review by sarcasticadmin — APPROVED** [review]
awesome, looks great @fossadmin :shipit:

**Pull request merged**